magic-modules/merge-pr.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -e
+set -x
+
+echo "$CREDS" > ~/github_private_key
+chmod 400 ~/github_private_key
+
+
+shopt -s dotglob
+cd mm-approved-prs
+ID=$(git config --get pullrequest.id)
+BRANCH=$(git config --get pullrequest.branch)
+cd ..
+cp -r mm-approved-prs/* mm-output
+
+cd mm-output
+git config pullrequest.id $ID
+git checkout $BRANCH
+git config --global user.email "nmckinley@google.com"
+git config --global user.name "Magic Module Bot"
+git config -f .gitmodules submodule.build/terraform.branch master
```

Comment:
When merging the PR I don't think you should use master. Instead you should point to the exact commit you just merged. Not doing so creates a racing condition that HEAD may move in between job tasks and the output in the terraform folder _may not_ reflect the execution of the MM compiler.

Reply:
I disagree, but this is a good place to argue about it.  :)  My argument is in `README.md` now - basically, you can't guarantee that downstream repos will merge PRs in the order they're produced, so there's a risk of jumping backwards if you point to the exact merge commit.

Reply:
Nice! You raise a great scenario. Let's sync about this offline as I think simply pointing to master will also have ill side effects. As of now I grant that my original suggestion of simply pointing to the hash is not sufficient, so we'll discuss a third option or fourth and report back here afterwards.